pkg/registry/autoscaling/horizontalpodautoscaler/strategy.go

```diff
@@ -71,6 +74,13 @@ func (autoscalerStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime
 	oldHPA := old.(*autoscaling.HorizontalPodAutoscaler)
 	// Update is not allowed to set status
 	newHPA.Status = oldHPA.Status
+
+	// Any changes to the spec increment the generation number, any changes to the
+	// status should reflect the generation number of the corresponding object.
+	// See metav1.ObjectMeta description for more information on Generation.
+	if !apiequality.Semantic.DeepEqual(oldHPA.Spec, newHPA.Spec) {
```

Comment:
Elsewhere (eg. deployments) we increment generation based on the annotations too

https://github.com/kubernetes/kubernetes/blob/808557e468639d8fd6027c99d004bceb8e8a5ec1/pkg/registry/apps/deployment/strategy.go#L106

Why should or should not this be the case here?

Reply:
Same question for all of the other updates.

Reply:
Yes, there are a couple of resources that have different behavior based on annotations/labels (eg. namespaces, nodes, potentially more).

Reply:
Making annotations part of generation isn't something we should propagate. That implies they should have been under spec.